lib/grandpa/network.go

```diff
@@ -63,8 +63,8 @@ func (*GrandpaHandshake) Hash() (common.Hash, error) {
 	return common.Hash{}, nil
 }
 
-// IsHandshake returns true
-func (*GrandpaHandshake) IsHandshake() bool {
+// IsValidHandshake returns true
+func (*GrandpaHandshake) IsValidHandshake() bool {
```

Comment:
would be nice to have a similar check here since the `GrandpaHandshake` should come from other authority peers
```go
switch hs.Roles {
	case common.AuthorityRole:
		return true
	default:
		return false
	}
```

Reply:
addressed